client/state/terms/selectors.js

```diff
@@ -5,14 +5,63 @@ import get from 'lodash/get';
 import values from 'lodash/values';
 
 /**
+ * Internal dependencies
+ */
+import {
+	getSerializedTermsQuery
+} from './utils';
+
+/**
+ * Returns true if currently requesting terms for the taxonomies query, or false
+ * otherwise.
+ *
+ * @param  {Object}  state  Global state tree
+ * @param  {Number}  siteId Site ID
+ * @param  {String}  taxonomy Taxonomy slug
+ * @param  {Object}  query  Taxonomy query object
+ * @return {Boolean}        Whether terms are being requested
+ */
+export function isRequestingTermsForQuery( state, siteId, taxonomy, query ) {
+	const serializedQuery = getSerializedTermsQuery( query );
+	return !! get( state.terms, [ 'queryRequests', siteId, taxonomy, serializedQuery ] );
```

Comment:
Not to hold up merge, but we know that `queryRequests` will always exist on `state.terms`, so we could move it out of the path argument (avoids unnecessary existence checking).